CASMTRIAGE-3173 updated PXE troubleshooting docs to include how to resolve PXE boot loop if found (#1552)
Cray-HPE/docs-csm: install/pxe_boot_troubleshooting.md

    - [Aruba Configuration](#aruba-configuration)
    - [Mellanox Configuration](#mellanox-configuration)
  - [Next steps](#next-steps)
    - [Node iPXE Retries and NIC Order](#pxe-nic-boot-order)
    - [Restart BSS](#restart-bss)
    - [Restart KEA](#restart-kea)
    - [Missing BSS Data](#missing-bss-data)

In order for PXE booting to work successfully, the management network switches need to be configured correctly.

<a name="#required-configuration"></a>

## Configuration required for PXE booting

To successfully PXE boot nodes, the following is required:

> - The IP helper-address must be configured on VLAN 1,2,4,7. This will be where the layer 3 gateway exists (Spine or Leaf).

Reviewer: ```suggestion
- The IP helper-address must be configured on VLANs 1, 2, 4, and 7. This will be where the layer 3 gateway exists (Spine or Leaf).
```

> - The virtual-IP/VSX/MAGP IP address must be configured on VLAN 1,2,4,7.

Reviewer: ```suggestion
- The virtual-IP/VSX/MAGP IP address must be configured on VLANs 1, 2, 4, and 7.
```

> - `spine01/spine02` needs an active gateway on VLAN1 this can be identified from `MTL.yaml` generated from CSI.

Reviewer: ```suggestion
- `spine01` and `spine02` need an active gateway on VLAN 1; this can be identified from `MTL.yaml` generated by CSI.
```

> - `spine01/spine02` needs an IP helper-address on VLAN1 pointing to 10.92.100.222.

Reviewer: ```suggestion
- `spine01` and `spine02` need an IP helper-address on VLAN 1 pointing to `10.92.100.222`.
```

    - [Aruba Configuration](#aruba-configuration)
    - [Mellanox Configuration](#mellanox-configuration)
  - [Next steps](#next-steps)

>     - [Node iPXE retries and NIC order](#pxe-nic-boot-order)

Reviewer: ```suggestion
    - [Node iPXE Retries and NIC Order](#pxe-nic-boot-order)
```

        ip helper-address 10.92.100.222
    ```

> 2. If any of this configuration is missing, update it to BOTH switches.

Reviewer: ```suggestion
2. If any of this configuration is missing, then update it on BOTH switches.
```

    interface vlan 1 magp 1 ip virtual-router mac-address 00:00:5E:00:01:01
    ```

> 1. If this configuration is missing, add it to BOTH switches.

Reviewer: ```suggestion
1. If this configuration is missing, then add it to BOTH switches.
```

> If the configuration looks good, and PXE boot is still not working, there are some other things to try.

Reviewer: ```suggestion
If the configuration looks good and PXE boot is still not working, then there are some other things to try.
```

> <a name="pxe-nic-boot-order"></a>

Reviewer: ```suggestion
<a name="pxe-nic-boot-order"></a>

```

> ### Node iPXE retries and NIC order

Reviewer: ```suggestion
### Node iPXE Retries and NIC Order
```

> In some environments you may find that ncn-m001 will be looping through all the NICs and still failing to PXE boot after the 3rd Chain attempt during the [Deploy Final NCN reboot step](https://github.com/Cray-HPE/docs-csm/blob/release/1.2/install/deploy_final_ncn.md#4-reboot). The order of NICs used by default is designed to be optimal for multiple types of hardware and cabling, but may need to be edited for your environment to reduce the time to boot ncn-m001. If so, follow the steps in the [Edit the iPXE Embedded Boot Script procedure](https://github.com/Cray-HPE/docs-csm/blob/release/1.2/operations/boot_orchestration/Edit_the_iPXE_Embedded_Boot_Script.md) to adjust the order of NICs used such that net0, or others, come before net2. If that does not resolve the issue, return to this PXE Boot Troubleshooting page.

Reviewer: ```suggestion
In some environments, during the [Deploy Final NCN reboot step](deploy_final_ncn.md#4-reboot), `ncn-m001` may loop through all of its NICs and still fail to PXE boot,
even after the 3rd chain attempt. The NIC boot ordering used by default is designed to be optimal for multiple types of hardware and cabling, but it may need to be
edited for specific environments in order to reduce the boot time of `ncn-m001`. If the boot issues described above are observed, then follow the steps in the
[Edit the iPXE Embedded Boot Script procedure](../operations/boot_orchestration/Edit_the_iPXE_Embedded_Boot_Script.md), adjusting the NIC boot order such that `net0`,
or others, come before `net2`. If that does not resolve the issue, then return to this PXE Boot Troubleshooting page.
```

1. Reboot to the PIT.

   - If using a USB PIT:

     1. Reboot the PIT node, watching the console as it boots.

     1. Manually stop it at the boot menu.

     1. Select the USB device for the boot.

     1. Once booted, log in and mount the data partition.

        ```bash
        pit# mount -vL PITDATA
        ```

   - If using a remote ISO PIT, follow the [Bootstrap LiveCD Remote ISO](bootstrap_livecd_remote_iso.md) procedure up through (**and including**) the [Set Up The Site Link](bootstrap_livecd_remote_iso.md#set-up-site-link) step.

1. Set variables for the system name, the CAN IP address for `ncn-m002`. the Kubernetes version, and the Ceph version.

    The CAN IP address for `ncn-m002` is obtained [at this step of the Deploy Final NCN procedure](deploy_final_ncn.md#collect-can-ip-ncn-m002).

    The Kubernetes and Ceph versions are from the output of the [`csi handoff ncn-images` command in the Deploy Final NCN procedure](deploy_final_ncn.md#ncn-boot-artifacts-hand-off).

>  If needed, the typescript file from that procedure should be on `ncn-m002` in the `/metal/bootstrap/prep/admin` directory.

Reviewer: ```suggestion
 If needed, the typescript file from that procedure should be on `ncn-m002` and `ncn-m003` in the `/metal/bootstrap/prep/admin` directory.
```
